### PR TITLE
Trev8939/raster layer file

### PR DESCRIPTION
**Pull request by TADraeseke**
Adds sample which demonstrates loading a raster from the local device.

### CODE REVIEW

**Comment by TADraeseke** [comment]
@doneill Just need a valid link to hold the Shasta.tif and associated world files, etc.

README currently has placeholder links

File: raster-layer-file/build.gradle [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+        exclude group: 'com.android.support', module: 'support-annotations'
+    })
+    compile 'com.android.support:appcompat-v7:25.2.0'
+    compile 'com.android.support.constraint:constraint-layout:1.0.2'
+    testCompile 'junit:junit:4.12'
+}
```
Comment (doneill) [inline raster-layer-file/build.gradle]: Clean up dependencies, only need `constraint-layout` at this level.

File: raster-layer-file/src/main/java/com/esri/arcgisruntime/sample/rasterlayerfile/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    /**
+     * Loads Shasta.tif as a Raster and adds it to a new RasterLayer. The RasterLayer is then added
+     * to the map as an operational layer. Map viewpoint is then set based on the raster's geometry.
+     */
+    private void loadRaster() {
+        // create a raster from a local raster file
+        Raster raster = new Raster(
+                new File("/storage/emulated/0/ArcGIS/samples/raster/Shasta.tif").getAbsolutePath());
```
Comment (doneill) [inline raster-layer-file/src/main/java/com/esri/arcgisruntime/sample/rasterlayerfile/MainActivity.java]: Avoid hard coding the path, use `Environment.getExternalStorageDirectory()` and the local path as a string resource.

File: raster-layer-file/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Raster Layer File
+
+Demonstrates how to create and use a raster layer made from a local raster file.
+
+![](raster-layer-file.png)
```
Comment (doneill) [inline raster-layer-file/README.md]: Added a new screenshot, 700x420.

File: raster-layer-file/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+- Create a <code>RasterLayer</code> from the `Raster`.
+- Add it as an operational layer with `map.getOperationalLayers().add(rasterLayer)`.
+
+
+## Features
+- Raster
+- RasterLayer
+
+
+
```
Comment (doneill) [inline raster-layer-file/README.md]: Converted from HTML to markdown content

File: raster-layer-file/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+6. Create the ArcGIS/samples directory, `mkdir ArcGIS/samples/raster`.
+7. You should now have the following directory on your target device, `/sdcard/ArcGIS/samples/raster`. We will copy the contents of the downloaded data into this directory. Note:  Directory may be slightly different on your device.
+8. Exit the shell with the, `exit` command.
+9. While still in your command prompt, navigate to the root folder where you extracted the contents of the data from step 1 and execute the following command: 
+	* `adb push raster-file/. /sdcard/ArcGIS/samples/raster`
+
+
+Link | Local Location
+---------|-------|
+|[raster-file.zip](https://arcgisruntime.maps.arcgis.com/home/item.html?id=7c4c679ab06a4df19dc497f577f111bd)| `<sdcard>`/ArcGIS/samples/raster/Shasta.tif |
```
Comment (doneill) [inline raster-layer-file/README.md]: Updated public link site url for data

File: raster-layer-file/README.md (hunk 3 above)
Comment (TADraeseke) [inline raster-layer-file/README.md]: Obviously your choice, but this screenshot is a little unclear to me.

It doesn't look to be taken from an handheld/Android device and doesn't hint at being georeferenced (by lining up with a basemap of some kind). My vote would be something more like the image in the [Java sample](https://github.com/Esri/arcgis-runtime-samples-java/blob/master/src/main/java/com/esri/samples/raster/raster_layer_file/RasterLayerFile.png).

**Comment by TADraeseke** [comment]
Cleaned gradle and made changes to build path via strings.xml values. Updates to readme look good to me, file downloads fine and all links work. 

See above for thoughts regarding screenshot.

File: raster-layer-file/README.md (hunk 3 above)
Comment (doneill) [inline raster-layer-file/README.md]: @TADraeseke Fair point, I updated the screenshot to match image dimensions (700x420).

**Comment by doneill** [comment]
@TADraeseke Updates look good to merge.

**Pull request merged**